### DISCUSSION TITLE
Implementierung ebus

### DISCUSSION THREAD

**Discussion by fred081273** [cyberthom42/vaillant-arotherm-plus#7] [General]
Mir würde es sehr helfen, wenn wir zusammentragen könnten, was man überhaupt umsetzen muss, um Werte in der Steuerung ändern zu können, bzw. Skripte zu implementieren.
Also eine Anleitung, welche Hard+ und Software man benötigt und wie man das implementieren kann. Dankeschön!

**Comment by RotorMotor42** [comment]
Um mit dem ebus interagieren zu können empfiehlt sich ebusd: https://ebusd.eu/
Dort ist auch beschrieben welche Hardware/Adapter man verwenden kann.
Mit dem ebusd kann man dann Werte über den Bus lesen und schreiben.

Was für konkrete Fragen hast du dazu?

**Reply by fred081273** [reply]
Danke, das hat schon Mal geholfen. Ich habe mich jetzt für einen WLAN Adapter registriert. Mal schauen, wie lange das dauert.
Hilfe werde ich wohl erst bei der Implementierung benötigen.

**Reply by cyberthom42** [reply]
Du solltest alternativ schon mal nach einem Adapter in Gebrauchtbörsen und Foren suchen. Der letzte Batch der Adapter hatte eine sehr lange Lieferzeit. Mit den aktuellen wird es sicherlich noch schlechter. Alternativ gibt es noch andere, wie z.B. den Esera. Wichtig: Der Adapter muss auf den Bus schreiben können.

**Comment by Tollentino** [comment]
Mein ebus Adapter von John kommt bald. Jetzt gibts ja schon bei John Anleitungen etc. Aber die gehen irgendwie alle davon aus, dass man schon weiß, was man mit nem Raspberry pi machen muss usw. Gibt es gute Tutorials (text nicht video) für Raspberry, Docker Container usw usf?
Danke und Gruß!

**Reply by cyberthom42** [reply]
Ich könnte dazu was ins Wiki schreiben. Deine Frage zeigt auch schon das "Problem". Man muß sich zunächst erst einmal entscheiden wie man es installieren möchte und vor allem wie es eingebunden werden soll. Ich habe auf meinem Raspi ein normales raspian laufen und ebus direkt installiert. Die Daten hole ich über iobroker, zudem habe ich mqtt aktiviert. 
Da ich das vor 1,5 Jahren eher auf die schnelle gemacht habe, wollte ich das irgendwann mal frisch installieren. Aber das Provisorium läuft bekanntlich ja immer länger als man denkt. Ich habe auf den Raspi bisher eigentlich nie den Bedarf für docker gesehen, ich habe einen separaten Server auf dem das was ich unter docker machen möchte nutze. Es geht aber auch auf dem raspi. Bringt halt zusätzlich ein paar Einschränkungen bezüglich commando Zeilen Zugriff usw. Bringt aber auch Vorteile beim Update. Wobei ich meine ebus Installation nie automatisch updaten lassen würde....
Was möchtest du denn machen mit dem ebusd?

**Reply by Tollentino** [reply]
Ok, ja, ich weiß tatsächlich gar nichts außer ein paar Buzzwords die ich aufgeschnappt hatte.

Erstmal nur messen und protokollieren (arotherm WP). Später evtl. auch steuern.

Da ich grundsätzlich auch noch meine Shelleys irgendwann über den raspi steuern möchte, dachte ich eine saubere Trennung über Docker wäre sinnvoll. wenn man das nicht braucht, lass ich das auch.

**Reply by cyberthom42** [reply]
Also wenn du messen möchtest, heißt das sicherlich du willst das in eine Datenbank schreiben und auswerten. Wenn du später dann auch steuern möchtest, solltest du dich also als erstes für ein SmartHome System oder eine andere Alternative entscheiden, mit der du das machst. Gängige Lösungen sind iobroker oder Home Assistant. Für die Anzeige der Grafiken bietet sich Grafana an. 
Sowohl iobroker als auch HomeAssistant bieten hier von Haus aus einfache Möglichkeiten die Werte in eine Datenbank zu schreiben. 
Die Installation von ebusd ist davon unabhängig. Hier muß man sich entscheiden, ob man alles über MQTT machen will oder mischen. Ich mische es aktuell.

**Comment by Tollentino** [comment]
Gibt es denn etwas, was man normalerweise machen wollte, was MQTT nicht kann?
Also genau, ich würde die Messdaten in ne Datenbank schreiben lassen. MySQL oder PostgreSQL. Ehrlich gesagt, würde mir für den Anfang sogar nen csv dump reichen. Analysieren würde ich wahrscheinlich dann in Excel oder PowerBI (kenne ich mich einfach besser aus).
Für später steuern, ja genau die beiden hatte ich schon auf der Uhr. Was hat denn davon die größere Kompatibilitätsbreite?

**Reply by cyberthom42** [reply]
Alles was in Richtung Debugging geht wird mit MQTT schwieriger. Ansonsten kann man per MQTT auch die gleichen Messages senden und empfangen. 
Bezüglich Datenbank wird es ja eher eine Time-Series Datenbank, also INFLUXDB oder ähnliches. 
Iobroker ist in Deutschland sehr verbreitet, HomeAssistant eher international. Ich habe jetzt keinen 1:1 Vergleich, aber die Kompatibilität sollte ähnlich sein. Es sind unterschiedliche Konzepte dahinter. Bei iobroker kann man recht individuell einfache Scripte (z.B. mit Blockly) erstellen. Wenn man mit üblichen "if-than-than-that" Themen auskommt, sind die glaube ich besser in HomeAssistant integriert. Ist aber denke ich auch eine Glaubensfrage. 
Wenn ich Zeit habe, werde ich mir auch mal parallel eine HomeAssistant Installtion auf den Server legen und schauen wie das so im Vergleich ist. Ich nutze iobroker schon viele Jahre und war immer sehr zufrieden.

**Reply by Tollentino** [reply]
puh, ich sehe, da habe ich viel nachhol/lernbedarf.
Ich habe gerade gesehen, dass John ja auch eine Erste Schritte-Anleitung hat. Vielleicht arbeite ich die erstmal durch wenn das Teil da ist.
Dort schreibt er übrigens auch dass es mit HomeAssistant möglich ist ebusd als Addondor laufen zu lassen. Evtl. ist das eine gut e Möglichkeit um nicht alles selbst einrichten zu müssen?

**Reply by chrizzzp** [reply]
Sehe im Prinzip alles genauso wie [cyberthom42](https://github.com/cyberthom42), kann selbst aber nur für HomeAssistant sprechen, wovon ich ziemlich begeistert bin. Installationsmöglichkeiten gibt es mehrere (von "Einsteiger-tauglich" bis "für Nerds"). Eigentlich wollte ich es nur mal testen und habe einen alten Intel-DualCore-Netbook dafür hergenommen, auf dem ich erst Debian-Linux und dann 'Home Assistant Supervised' installiert habe (nachdem der alte Akku entfernt und die HDD durch eine SSD ersetzt wurde). Der Netbook verbraucht (mit ausgeschaltetem Display) ca. 10 W (ein Raspi ist natürlich nochmal etwas sparsamer). Auf auf dem Netbook (jetzt im Keller) läuft mein System (HA, ebusd, MQTT, Grafana und noch ein paar weitere Addons) noch immer... (hätte nie geglaubt, daß ich für den alten Rechner nochmal Verwendung habe).

Ganz gute Video-Einführungen für HomeAssistant gibt es auf Youtube, eine Empfehlung wären z.B. die (deutschen) Videos von Simon42. Wenn's ins Detail geht muß man sich in jedes System aber trotzdem einlesen/einarbeiten. Die HomeAssistant-Foren sind größtenteils in Englisch, wenn das keine Hürde ist, findet man sehr viel Information im Netz.

Für HomeAssistant (außer die reine "Core-Installation") kann man mittels HA-Supervisor sehr bequem Addons installieren, wie z.B. das ebusd-Addon von Lukas Grebe (https://github.com/LukasGrebe/ha-addons), das die Installationen und Konfiguration von ebusd sehr unkompliziert macht (automatisiert). Das gilt auch für das Mosquitto-Addon, den MQTT-Broker,  über den man bequem mit ebusd aus HA kommunizieren kann. Nach ein paar Anfangsschwierigkeiten, funktioniert auch letzteres sehr gut.

**Reply by Tollentino** [reply]
ok, ich schau mir das mal am Wochenende an. Der ebus-adapter sollte die Woche ankommen...

**Reply by cyberthom42** [reply]
Meine Empfehlung: Das wichtigste ist sich mit dem ebusd und den csv Configs auseinandersetzen, vor allem welche Werte du wie oft auslesen lassen möchtest. Du kannst nicht alles auslesen. Bzw. erhöht sich das Interval zwischen den einzelnen Messwerten je mehr Werte du haben möchtest. Die Priorität stellst du ja mit r1, r2 usw ein. Am besten ist möglichst erst einmal alles zu nutzen was sowieso über den Bus kommt, siehe dazu auch den anderen Thread hier.
Zu HomeAssistant: Ich hatte das mal (Home Assistant OS) auf die schnelle auf meinem ESXi installiert. Sieht prinzipiell sehr gut aus und ist auch benutzerfreundlich. Was mich auf dem ersten Blick stört sind die fehlenden Einstellmöglichkeiten. Aber das ist dann wahrscheinlich gleichzeitig auch ein Feature :-) Bei iobroker kann ich in der Nutzeroberfläche z.B. für jedes einzelne Objekt einstellen, was in die Datenbank geschrieben werden soll und wie oft (Interval). Bei HomeAssistant muss man das so wie ich es sehe dann per Config-File machen. Aber viele Einstellungen sind dafür recht intiutiv bedienbar wenn man nicht so tief eingreifen möchte.

**Reply by chrizzzp** [reply]
Für ebusd muß ein (einsteigerfreundliches) Tutorial m.E. erst noch geschrieben werden. Da könnte man sich noch Lorbeeren verdienen... ;-)

Die Wiki-Seiten von ebusd (https://github.com/john30/ebusd/wiki/) sind zwar technisch sehr informativ und gehaltvoll, aber nicht gerade was für Einsteiger, wie ich finde... Wie ich das mitbekommen habe, hat sich jeder hier die Infos dazu wohl aus dem Netz zusammengesucht. Wie man auch den Diskussionen auf der ebusd-Github-Seite entnehmen kann, tauchen auch immer wieder die gleichen Fragen und Probleme am Anfang auf (also nicht ins Bockshorn jagen lassen, ist normal... ;-)).

Ich empfehle übrigens gleich von Beginn an auf lokale CSV-Dateien zu setzen (also die auf dem eigenen System liegen), die man an sein System anpassen kann und woran man das Konzept besser erlernen kann. Man braucht ja auch nur einen Bruchteil der CSV-Dateien, die zum Repository gehören.
